fix(normal): fix repeated trigger modechanged for scheduled callback

MunifTanjim opened the pull request:
Resolves https://github.com/neovim/neovim/issues/22263

MunifTanjim committed:
test: scheduled callback shouldn't trigger ModeChanged repeatedly

zeertzjq commented:
I just found another problem that also applies to Vim. Created a PR at https://github.com/vim/vim/pull/12298

zeertzjq committed:
vim-patch:9.0.1490: the ModeChanged event may be triggered too often

Problem:    The ModeChanged event may be triggered too often.
Solution:   Only trigger ModeChanged when no operator is pending.
            (closes vim/vim#12298)

https://github.com/vim/vim/commit/73916bac5ac2a054a0c71adfe8d742691cdfd95c
